python/CCLE_data.py

```diff
@@ -38,29 +42,123 @@ def common_celllines(a,b):
     ax.set(xlabel = 'Histone Markers')
     plt.show
+    
+def pearson_dfs(gene_list, histone_list, data1, data2):
+    """This functions purpose is to create two dataframes. The first dataframe will contain all the r-values between the two sets of data. The second will contain all the p-values for each r-value.
+    
+    This function is very specific. The dataframe created will contain histone markers as the columns and gene names as the index. 
+    
+    This function assumes you already took the intersection between the two arrays. If you do not do that, then the matrix dimensions will not agree, resulting in an error.
+    
+    The format for inputting matrix should have columns correspond to the celllines.
+    """
+    Rmat = (len(gene_list), len(histone_list))
+    Rmat = np.zeros(Rmat)
+    Pmat = Rmat[:]
+    i = 0 
+    for gene in gene_list:
+        j = 0
+        for histone in histone_list:
+            data1.loc[gene] = data1.loc[gene].fillna(method='ffill')
+            data2.loc[histone] = data2.loc[histone].fillna(method='ffill')
+            correlation = pearsonr(data1.loc[gene], data2.loc[histone])
+            Rmat[i][j] = correlation[0]
+            Pmat[i][j]
+            j = j+1
+        i = i+1
+        
+    Rdf = pd.DataFrame(Rmat)
+    Rdf.columns = histone_list
+    Rdf.insert(0, 'Genes', gene_list, True)
+    Rdf = Rdf.set_index('Genes')
+    
+    Pdf = pd.DataFrame(Pmat)
+    Pdf.columns = histone_list
+    Pdf.insert(0, 'Genes', gene_list, True)
+    Pdf = Pdf.set_index('Genes')
+    return Rdf, Pdf
+
+
+
+    
+def Remove(duplicate): 
+    """This function will remove duplicated elements from lists"""
+    final_list = [] 
+    for num in duplicate: 
+        if num not in final_list: 
+            final_list.append(num) 
+    return final_list
+
+def Clustermap(data, size, colour, method, metric):
+    plt.figure(figsize=size)
+    ax = sns.clustermap(data, cmap = colour, method = method, metric= metric)
+    plt.show
+    
+def PlotlyHeat(df, size, title, xaxis, yaxis):
+    pio.renderers.default = "chrome"
+    fig = go.Figure(
+    data=(go.Heatmap(z = df, x =xaxis, y = yaxis, colorscale = "rdbu")),
+    layout_title_text=title)
+    
+    if size == None:
+        fig.update_layout(
+                autosize=True)
+    else:
+        fig.update_layout(
+                autosize=False,
+                width = size[0],
+                height = size[1])
+
+    fig.show()
+    
+
+
+"""
+"""
+
+"""
+"""
+
+def TissueAnalysis(dictionary, common_cellline, data1, data2, name):
+    df1 = data1.copy()
+    df2 = data2.copy()
+    for celllines in common_cellline:
+        if dictionary[celllines] != name:
+            del df1[celllines]
+            del df2[celllines]
+    matrix = Matrix(search, h3_markers, df1, df2)
+    matrix = pd.DataFrame(matrix)
+    matrix.columns = h3_markers
+    matrix.insert(0, 'Genes', search , True)
+    matrix = matrix.set_index('Genes')
+    return matrix
 
 """
```

Comment:
I don't know what this code is for. If you're cleaning up the dataset, you may want to move this to another script.